llvm/lib/Target/Mips/AsmParser/MipsAsmParser.cpp

```diff
@@ -2108,6 +2108,10 @@ bool MipsAsmParser::processInstruction(MCInst &Inst, SMLoc IDLoc,
   if ((Opcode == Mips::JAL || Opcode == Mips::JAL_MM) && inPicMode()) {
     warnIfNoMacro(IDLoc);
 
+    if (!Inst.getOperand(0).isExpr()) {
+      return Error(IDLoc, "jal needs a symbolic expression in PIC mode");
```

Comment:
gas gives the error message like:
```
k.s: Assembler messages:
k.s:2: Error: unsupported constant in relocation
```
I think that we should keep them same.

Reply:
sounds good, updated to the same message, thank you!